plugins/carbon_black_defense/komand_carbon_black_defense/actions/find_event/action.py

```diff
@@ -1,68 +1,68 @@
+        enriched_event_type = params.get(Input.ENRICHED_EVENT_TYPE)
+        process_hash = params.get(Input.PROCESS_HASH)
+        device_name = params.get(Input.DEVICE_NAME)
+        time_range = params.get(Input.TIME_RANGE)
 
-        if payload:
-            result = requests.get(url, headers=headers, params=payload)
-        else:
-            result = requests.get(url, headers=headers)
-        try:
-            data = komand.helper.clean(result.json())
-        except ValueError:
-            self.logger.error(result.text)
-            raise Exception(
-                f"Error: Received an unexpected response"
-                f" (non-JSON or no response was received). Raw response in logs."
-            )
-        if result.status_code == 200:
-            return {
-                Output.SUCCESS: data["success"],
-                Output.LATESTTIME: data["latestTime"],
-                Output.RESULTS: data["results"],
-                Output.ELAPSED: data["elapsed"],
-                Output.MESSAGE: data["message"],
-                Output.TOTALRESULTS: data["totalResults"],
-            }
-        if result.status_code in range(400, 499):
-            raise Exception(
-                f"Carbon Black returned a {result.status_code} code."
-                f" Verify the token and host configuration in the connection. Response was: {result.text}"
-            )
-        if result.status_code in range(500, 599):
-            raise Exception(
-                f"Carbon Black returned a {result.status_code} code."
-                f" If the problem persists please contact support for help. Response was: {result.text}"
+        criteria = {}
+
+        if device_external_ip:
+            criteria["device_external_ip"] = device_external_ip
+        if process_name:
+            criteria["process_name"] = process_name
+        if enriched_event_type:
+            criteria["enriched_event_type"] = enriched_event_type
+        if process_hash:
+            criteria["process_hash"] = process_hash
+        if device_name:
+            criteria["device_name"] = device_name
+        if len(criteria) == 0:
+            raise PluginException(
+                cause="Error. Have not entered a criteria for action to run.",
+                assistance="Enter a criteria of at least one plugin input.",
             )
-        self.logger.error(result.text)
-        raise Exception(
-            f"An unknown error occurred."
-            f" Carbon Black returned a {result.status_code} code. Contact support for help. Raw response in logs."
-        )
+        id_ = self.connection.get_job_id_for_enriched_event(criteria, None, time_range)
+
+        self.logger.info(f"Got enriched event job ID: {id_}")
+        if id_ is None:
+            return {Output.EVENTINFO: None, Output.SUCCESS: False}
+        enriched_event_search_status = self.connection.get_enriched_event_status(id_)
+
+        t1 = datetime.now()
+        for _ in range(0, 9999):
```

Comment:
Is it a backoff? I've seen that same thing somewhere in our codebase, not the prettiest bit. I'd change it to something like: [back off](https://stackoverflow.com/a/5242013) We should probably add it to utils as a decorator.

Reply:
I've updated this in the lastest commit and added a new retry method.